torch/testing/_internal/common_methods_invocations.py

```diff
@@ -17043,6 +17043,20 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
         torch_opinfo_variant_name="with_dtype",
         supports_out=False,
     ),
+    ElementwiseUnaryPythonRefInfo(
+        "_refs.nn.functional.softsign",
+        torch_opinfo_name="nn.functional.softsign",
+        supports_nvfuser=False,
+        skips=(
+            DecorateInfo(
+                unittest.skip("Skipped!"), 'TestCommon', 'test_python_ref', dtypes=(torch.int8,)),
```

Comment:
Why are these skipped?

Reply:
Hi @kshitij12345, these tests were skipped because of their relative & absolute differences. Thanks for asking. I added a comment there. :)

Reply:
`softsign` is implemented in Python, so it's kind of its own reference:

https://github.com/pytorch/pytorch/blob/2d73c8e6e0378655f732a48ec50ae1908ce0a4a4/torch/nn/functional.py#L1733

I think the computation it performs is distinct from the computation in this reference, however. In the original, the type promotion is implicit, so I think `abs(a) + 1` occurs in the datatype of `a`. In this reference the type promotion is explicit, so `abs(a) + 1` occurs in the promoted datatype. Does this account for the discrepancy between the operations? Note that the order of the arguments is distinct, too, and `a + b` may not equal `b + a`.

So let's first understand where the discrepancies are coming from, and then let's think how best to align the original PyTorch operation and its reference. What's particularly surprising to me is that the reference is not detected as "more accurate" by `test_python_ref`. I thought that test would re-run operations in higher precision, and I would expect that the reference's type promotion behavior would result in a more numerically accurate computation.

Reply:
cc @lezcano as he might appreciate this

Reply:
We should either fix eager implementation, or disable support for int8 in eager, because eager does accidental type promotion. `test_python_ref` doesn't detect reference as more accurate because for integral types `test_python_ref` retains the datatype for "more accurate" computation, it's generally the right thing to do, except for accidental type promotion cases like here. And, for everyone's entertainment, here's `a.abs()+1` for int8 inputs, results are very much like you would think they would be, positive and larger than input:
```
In [1]: import torch

In [2]: a=torch.tensor([-128, 127], dtype=torch.int8)

In [3]: a.abs()+1
Out[3]: tensor([-127, -128], dtype=torch.int8)
```

Reply:
It's funny that in this case we probably want an operation that says "move one away from zero" before the `abs`. 

What's the problem with promoting to int64 in the more accurate test? Shouldn't the addition occur as expected then?

Reply:
No problem, but then we wouldn't catch glaring error in eager.

Reply:
From offline discussion: @ngimel and I agree the real error here was allowing boolean and integer inputs in the first place

Reply:
One part of me says we should fix this, but then you have the people that do imaging, starting with the whole torchvision team, that represent the images with `uint8`s and want fast operations on `uint8`s, whatever that may even mean.

All this is to say that I don't think we can safely remove the support for integral dtypes for this op. The correct approach, which would be a *massive* pain to implement, would be to deprecate the integer support for any function that doesn't take the integers to the integers. Now, who'd dare pulling such a feat, that I don't know.

Reply:
Perhaps @mruberry or @ngimel have a different opinion, but I think we should either (or both):
- Strenghten the preconditions of `softsign` and disallow passing ints / bools
- Drop this PR, as that function is already implemented in Python

Reply:
I suggest option 2 and punting on improving softsign at this time